syntax = "proto3";


service Repository {
    rpc InitializeRepository(InitializeRepositoryRequest) returns (InitializeRepositoryResponse) {}; 
}


message InitializeRepositoryRequest {
    map<string, bytes> files = 1;
    string session_id = 2;
    int32 batch_number = 3;
    int32 num_total_batches = 4;
}

message InitializeRepositoryResponse {
    string repository_id = 1;
}
